### PR TITLE
Sign release artifacts

### PR DESCRIPTION
**Pull request by arthurpassos**
### Changelog category (leave one):
- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Calculates the hash, sign and post all release artifacts to to GH storage/ S3.

Files hashed are fetched from uploaded builds artifacts filtered by the `build_name`, which corresponds to `package_release`. `package_release` contains the build artifacts generated by `BuilderDebRelease`.

Signed hashes:
```
arthur@arthur:~/Downloads/signed_hashes_dir$ ls
clickhouse-client_22.3.15.2.altinitystable_amd64.deb.sha512.gpg              clickhouse-keeper_22.3.15.2.altinitystable_x86_64.apk.sha512.gpg
clickhouse-client-22.3.15.2.altinitystable-amd64.tgz.sha512.gpg              clickhouse-keeper-22.3.15.2.altinitystable.x86_64.rpm.sha512.gpg
clickhouse-client_22.3.15.2.altinitystable_x86_64.apk.sha512.gpg             clickhouse-keeper-dbg_22.3.15.2.altinitystable_amd64.deb.sha512.gpg
clickhouse-client-22.3.15.2.altinitystable.x86_64.rpm.sha512.gpg             clickhouse-keeper-dbg-22.3.15.2.altinitystable-amd64.tgz.sha512.gpg
clickhouse-common-static_22.3.15.2.altinitystable_amd64.deb.sha512.gpg       clickhouse-keeper-dbg_22.3.15.2.altinitystable_x86_64.apk.sha512.gpg
clickhouse-common-static-22.3.15.2.altinitystable-amd64.tgz.sha512.gpg       clickhouse-keeper-dbg-22.3.15.2.altinitystable.x86_64.rpm.sha512.gpg
clickhouse-common-static_22.3.15.2.altinitystable_x86_64.apk.sha512.gpg      clickhouse-keeper.sha512.gpg
clickhouse-common-static-22.3.15.2.altinitystable.x86_64.rpm.sha512.gpg      clickhouse-library-bridge.sha512.gpg
clickhouse-common-static-dbg_22.3.15.2.altinitystable_amd64.deb.sha512.gpg   clickhouse-odbc-bridge.sha512.gpg
clickhouse-common-static-dbg-22.3.15.2.altinitystable-amd64.tgz.sha512.gpg   clickhouse-server_22.3.15.2.altinitystable_amd64.deb.sha512.gpg
clickhouse-common-static-dbg_22.3.15.2.altinitystable_x86_64.apk.sha512.gpg  clickhouse-server-22.3.15.2.altinitystable-amd64.tgz.sha512.gpg
clickhouse-common-static-dbg-22.3.15.2.altinitystable.x86_64.rpm.sha512.gpg  clickhouse-server_22.3.15.2.altinitystable_x86_64.apk.sha512.gpg
clickhouse-keeper_22.3.15.2.altinitystable_amd64.deb.sha512.gpg              clickhouse-server-22.3.15.2.altinitystable.x86_64.rpm.sha512.gpg
clickhouse-keeper-22.3.15.2.altinitystable-amd64.tgz.sha512.gpg              clickhouse.sha512.gpg

```

Testing:
I have manually downloaded the artifacts ZIP and verified the signature of one of the files with `gpg --verify file.gpg`. Plus, I have changed one single byte of this file and tried the very same command, it failed with invalid signature or smth like that.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: tests/ci/sign_release.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    BLOCK_SIZE = 65536 # The size of each read from the file
+
+    file_hash = hashlib.sha256() # Create the hash object, can use something other than `.sha256()` if you wish
+    with open(file_path, 'rb') as f: # Open the file to read it's bytes
+        fb = f.read(BLOCK_SIZE) # Read from the file. Take in the amount declared above
+        while len(fb) > 0: # While there is still data being read from the file
+            file_hash.update(fb) # Update the hash
+            fb = f.read(BLOCK_SIZE) # Read the next block from the file
+
+    hash_file_path = file_path + '.sha512'
```
Comment (Enmk) [inline tests/ci/sign_release.py:24]: Why is it `.sha512` in file name, while we use `sha256` ?

**Review by Enmk — COMMENTED** [review]
Could you please specify which files are uploaded to s3? It is hard to deduce from the code

File: tests/ci/sign_release.py (hunk 1 above)
Comment (arthurpassos) [reply tests/ci/sign_release.py:24]: Because it used to be `sha512` and I forgot to update the filename lol, will update it

**Comment by arthurpassos** [comment]
> Could you please specify which files are uploaded to s3? It is hard to deduce from the code

You mean add a comment stating that build files are uploaded? Or literally specify all filenames?

**Pull request merged**